Use Case Docs: Headless CMS (#294)
directus/docs: cookbook/data-models/create-reusable-page-components-using-m2a.md

> ---

Reviewer: Ok on reading this again, it's a super solid piece of work, but it isn't and shouldn't be a recipe as far as I'm concerned - this is a fully fledged guide on getting folks to create a basic project. Fortunately, I also have the guides written as part of this packet. So let's keep the content as-is, and just shift this from a recipe to a guide. We'll action this together.

description: This recipe explains how build re-usable page components using Directus Many-to-Any (M2A) relationships.
tags: []
skill_level:
directus_version: 9.21.4
author_override:
author: Bryant Gillespie
---

> # Create re-usable page components

Reviewer: Title-case this

> {{ $frontmatter.description }}

:::tip Author: {{$frontmatter.author}}

<!-- **Skill Level:** {{$frontmatter.skill_level}}\ -->

**Directus Version:** {{$frontmatter.directus_version}}

<!-- **Tags:** {{$frontmatter.tags.join(", ")}} -->

:::

## Explanation

Many websites are made of common, repeating sections or groups of content.

> A common use case when using Directus as a headless CMS’s is creating individual blocks that can be re-used on many

Reviewer: ```suggestion
A common use case when using Directus as a headless CMS is creating individual blocks that can be re-used on many
```

different pages.

> This enables your content team create unique page layouts without having to use development resources whenever a new
> page layout is needed.
> 
> In this recipe, we’ll cover the steps needed to create re-usable page components inside your Directus project. Here’s
> the quick outline:

Reviewer: ```suggestion
This enables your content team create unique page layouts from re-usable components. To achieve this, you will:
```

- Mapping your data model
- Creating individual page blocks
- Creating your page collection
- Building pages with blocks
- Fetching page data from the API
- Working with your front-end

## How-To Guide

:::tip Requirements

You’ll need to have either a Directus Cloud project configured and running or a self-hosted instance of Directus up and
running.

:::

> ### Map out your data model

Reviewer: Title-case

> Before you starting creating Collections inside Directus, it’s helpful to map out your data model (schema) outside of
> Directus first.

Reviewer: ```suggestion
Before creating Collections inside of your Directus project, it’s helpful to think about your data model (schema).
```

Reviewer: From the first half, it's implied that they'll be doing this outside of the platform

Consider this sample page below.

![Sample Page](/headlesscms/sample-page.png)

There are three main “blocks” that could be broken down into separate components.

1. A hero block at the top of the page that includes a strong headline, an image, and some copy with a call to action.
2. A block of cards that could link out to blog posts or other content.
3. A block of rich text or HTML content.

> Let’s break down our data model for each section.

Reviewer: ```suggestion
Let’s break down the data model for each section.
```
